### ISSUE TITLE
Set-AzAksCluster cmdlet fails on subscription with many resources

### ISSUE THREAD

**Issue by bpinheiro19** [Azure/azure-powershell#21846]
### Description

In subscriptions with a large number of resources(1000+), the cmdlet Set-AzAksCluster feature to attach an ACR to an AKS cluster fails to retrieve an existing ACR and causes a failure in the command.
Attaching an ACR using azure cli command works well.

Command used: Set-AzAksCluster -Name `<AKS-name>` -ResourceGroupName `<RG-name>` -AcrNameToAttach `<ACR-name>` -Debug

The command will make the following http request: 
_https://management.azure.com/subscriptions/XXXXXXXXXX/resources?$filter=resourceType eq
'Microsoft.ContainerRegistry/registries' and name eq '`<ACR-name>`'&api-version=2016-09-01_

and return the following http response:
_Body:
{
  'value': [],
  'nextLink':
'https://management.azure.com/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resources?%24filter=resourceType+eq+%2
7Microsoft.ContainerRegistry%2fregistries%27+and+name+eq+%27`<ACR-name>`%27&api-version=2016-09-01&%24skiptoken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
}_

The command will fail with the error message: 
_Set-AzAksCluster : Could not find specified Acr '`<ACR-name>`' to attach._

### Issue script & Debug output

```PowerShell
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

https://management.azure.com/subscriptions/XXXXXXXXXX/resources?$filter=resourceType eq
'Microsoft.ContainerRegistry/registries' and name eq '`<ACR-name>`'&api-version=2016-09-01

DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Body:
{
  'value': [],
  'nextLink':
'https://management.azure.com/subscriptions/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/resources?%24filter=resourceType+eq+%2
7Microsoft.ContainerRegistry%2fregistries%27+and+name+eq+%27`<ACR-name>`%27&api-version=2016-09-01&%24skiptoken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
}

Confirm
Could not find specified Acr '`<ACR-name>`' to attach.
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is 'Y'): A
Set-AzAksCluster : Could not find specified Acr '`<ACR-name>`' to attach.
```

### Environment data

```PowerShell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.4
PSEdition                      Core
GitCommitId                    7.3.4
OS                             Linux 5.4.0-1107-azure #113~18.04.1-Ubuntu SMP Tue Apr 25 13:58:29 UTC 2023
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
> Get-Module Az*

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     2.12.2                Az.Accounts                         {Add-AzEnvironment, Clear-AzConfig, Clear-AzContext, Clear-AzDefault…}
Script     5.4.0                 Az.Aks                              {Disable-AzAksAddOn, Enable-AzAksAddOn, Get-AzAksCluster, Get-AzAksNodePool…}
Script     5.7.1                 Az.Compute                          {Add-AzImageDataDisk, Add-AzVhd, Add-AzVMAdditionalUnattendContent, Add-AzVMDataDisk…}
Script     5.7.0                 Az.Network                          {Add-AzApplicationGatewayAuthenticationCertificate, Add-AzApplicationGatewayBackendAddressPool, Add-A…
Script     6.6.1                 Az.Resources                        {Export-AzResourceGroup, Export-AzTemplateSpec, Get-AzDenyAssignment, Get-AzDeployment…}
Script     5.6.0                 Az.Storage                          {Add-AzRmStorageContainerLegalHold, Add-AzStorageAccountManagementPolicyAction, Add-AzStorageAccountN…
Script     1.1.1                 Az.Tools.Predictor                  {Disable-AzPredictor, Enable-AzPredictor, Open-AzPredictorSurvey, Send-AzPredictorRating}
Script     0.0.0.10              AzureAD.Standard.Preview            {Add-AzureADApplicationOwner, Add-AzureADDeviceRegisteredOwner, Add-AzureADDeviceRegisteredUser, Add-…
Script     0.9.3                 AzurePSDrive
```

### Error output

```PowerShell
> Resolve-AzError
WARNING: Upcoming breaking changes in the cmdlet 'Resolve-AzError' :
The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this alias to use `Resolve-AzError` instead.
Note : Go to https://aka.ms/azps-changewarnings for steps to suppress this breaking change warning, and other information on breaking changes in Azure PowerShell.

   HistoryId: 1

Message        : Could not find specified Acr '`<ACR-name>`' to attach.
StackTrace     :    at Microsoft.Azure.Commands.Aks.CreateOrUpdateKubeBase.AddAcrRoleAssignment(String acrName, String acrParameterName, AcsServicePrincipal 
                 acsServicePrincipal)
                    at Microsoft.Azure.Commands.Aks.SetAzureRmAks.<>c__DisplayClass40_0.<ExecuteCmdlet>b__0()
                    at Microsoft.Azure.Commands.Aks.KubeCmdletBase.RunCmdLet(Action action)
                    at Microsoft.Azure.Commands.Aks.SetAzureRmAks.ExecuteCmdlet()
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.<>c__3`1.<ExecuteSynchronouslyOrAsJob>b__3_0(T c)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet, Action`1 executor)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.CmdletExtensions.ExecuteSynchronouslyOrAsJob[T](T cmdlet)
                    at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()
Exception      : Microsoft.Azure.Commands.Common.Exceptions.AzPSArgumentException
InvocationInfo : {Set-AzAksCluster}
Line           : Set-AzAksCluster -Name `<AKS-name>` -ResourceGroupName `<RG-name>` -AcrNameToAttach `<ACR-name>`
Position       : At line:1 char:1
                + Set-AzAksCluster -Name `<AKS-name>` -ResourceGroupName `<RG-name>` …
                 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 1
```

**Comment by isra-fel** [comment]
Thanks for reporting. We are looking into it.

**Comment by YanaXu** [comment]
We're discussing this issue offline since there is sensitive data.

**Comment by YanaXu** [comment]
The request sent from Azure PowerShell to ARM looks like this (**no $top parameter**):
```
GET https://management.azure.com/subscriptions/<subscriptionId>/resources?$filter=<filter>&api-version=2016-09-01
```

The response sent from ARM to Azure PowerShell looks like this (**0 results with 'nextLink'**):
```json
{
  'value': [],
  'nextLink':'...'
}
```

According to [stable/2016-09-01/resources.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Resources/stable/2016-09-01/resources.json):
```json (**if $top is null, returns all resources.**)
{
            "name": "$top",
            "in": "query",
            "required": false,
            "type": "integer",
            "format": "int32",
            "description": "The number of results to return. If null is passed, returns all resources."

}
```
But [pagination](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#pagination) says:
```
Server may return less records than requested with nextLink. Returning zero records with nextLink is an acceptable response.
Clients must fetch records until the nextLink is not returned or is null. Clients should never rely on number of returned records to determinate if pagination is completed.
```

I will fix this issue in Set-AzAksCluster.

**Comment by YanaXu** [comment]
This issue is fixed in #22001 and will be released in [July 2023 (2023-07-04)](https://github.com/Azure/azure-powershell/milestone/148).
